modules/node_groups/templates/userdata.sh.tpl

```diff
@@ -4,3 +4,6 @@
 ${pre_userdata}
 
 sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
+%{ if run_bootstrap_script }
```

Comment:
why this is optional and rely on ami_id?

how this differ from amazon delivered AMI (how they launching bootstrap part?), asking as I never checked this and just using AWS AMIs

Reply:
I don't now exactly how it works, but I think that if you don't specify a custom AMI, then EKS merges your init scripts with their snippet that does the bootstrap process. However, when you use a custom AMI, they expect you to provide a fully-working init script that includes the bootstrapping section

Reply:
Yes, @jaimehrubiks is totally right. Here is my comment about this topic - https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1473#issuecomment-884869743.